site/lib/src/components/lint_filter_search_section.dart

```diff
@@ -0,0 +1,202 @@
+import 'package:jaspr/jaspr.dart';
+import 'package:universal_web/web.dart' as web;
+
+import 'chip.dart';
+import 'search.dart';
+
+enum RuleSetOption {
+  inFlutter('Flutter'),
+  inRecommended('Recommended'),
+  inCore('Core');
+
+  final String label;
+  const RuleSetOption(this.label);
+}
+
+class LintInfo {
+  LintInfo({
+    required this.name,
+    required this.hasFix,
+    required this.stable,
+    required this.inCore,
+    required this.inRecommended,
+    required this.inFlutter,
+  });
+
+  final String name;
+  final bool hasFix;
+  final bool stable;
+  final bool inCore;
+  final bool inRecommended;
+  final bool inFlutter;
+}
+
+@client
+class LintFilterSearchSection extends StatefulComponent {
+  const LintFilterSearchSection();
+
+  @override
+  State createState() => _LintFilterSearchSectionState();
+}
+
+class _LintFilterSearchSectionState extends State<LintFilterSearchSection> {
+  bool onlyFixable = false;
+  bool onlyStable = false;
+
+  RuleSetOption? selectedRuleSet;
+
+  String searchQuery = '';
+
+  final lintCards = <web.HTMLElement>[];
+  final lintsInfo = <LintInfo>[];
+
+  @override
+  void initState() {
+    super.initState();
+
+    if (kIsWeb) {
+      final cards = web.document
+          .getElementById('lint-cards')
+          ?.querySelectorAll('.card');
+      if (cards == null) return;
+
+      bool hasData(web.HTMLElement elem, String name) =>
+          elem.attributes.getNamedItem('data-$name')?.value == 'true';
+
+      for (var i = 0; i < cards.length; i++) {
+        final card = cards.item(i) as web.HTMLElement;
+        lintCards.add(card);
+
+        final lintName = card.id;
+        if (lintName.isEmpty) return;
+
+        lintsInfo.add(
+          LintInfo(
+            name: lintName,
+            hasFix: hasData(card, 'has-fix'),
+            stable: hasData(card, 'stable'),
+            inCore: hasData(card, 'in-core'),
+            inRecommended: hasData(card, 'in-recommended'),
+            inFlutter: hasData(card, 'in-flutter'),
+          ),
+        );
+      }
+    }
+  }
+
+  void updateFilters(void Function() callback) {
+    setState(() {
+      callback();
+    });
+
+    final lintsToShow = <String>{};
+
+    for (final lint in lintsInfo) {
+      final lintName = lint.name;
+      if (!lintName.contains(searchQuery)) continue;
```

Comment:
To keep the query case insensitive and gracefully handle extra spaces:

```suggestion
      if (!lintName.contains(searchQuery.trim().toLowerCase())) continue;
```